pkg/playback/http_replayer.go

```diff
@@ -0,0 +1,198 @@
+			var evt stripeEvent
+
+			if err != nil {
+				fmt.Printf("ERROR when forwarding webhook requests: %v", err)
+				continue
+			}
+			err = json.Unmarshal(webhookReq.Body, &evt)
+			if err != nil {
+				fmt.Printf("ERROR when forwarding webhook requests: %v", err)
+				continue
+			}
+
+			fmt.Printf("	> Forwarding webhook [%v].\n", evt.Type)
+			fmt.Printf("	> Received %v from client. Expected %v.\n\n", resp.StatusCode, expectedResp.StatusCode)
+			if err != nil {
+				fmt.Printf("ERROR when forwarding webhook requests: %v", err)
+				continue
+			}
+		}
+	}()
+}
+
+// returns error if something doesn't match the cassette
+func (httpReplayer *replayServer) getNextRecordedCassetteResponse(request *httpRequest) (response *httpResponse, err error) {
+	// the passed in request arg may not be necessary
+	uncastResponse, err := httpReplayer.replayer.write(request)
+	if err != nil {
+		return &httpResponse{}, err
+	}
+
+	wrappedResponse := (*uncastResponse).(httpResponse)
+
+	return &wrappedResponse, err
+}
+
+// Reads any contiguous set of webhook recordings from the start of the cassette
+// The caller must call response.Body().Close() on each of the returned http.Response's, otherwise there will be a resource leak
+func (httpReplayer *replayServer) readAnyPendingWebhookRecordingsFromCassette() (webhookRequests []*httpRequest, webhookResponses []*httpResponse, err error) {
+	webhookBytes := make([]cassettePair, 0)
+
+	// --- Read the pending webhook interactions (stored as raw bytes) from the cassette
+	for httpReplayer.replayer.interactionsRemaining() > 0 {
+		interaction, err := httpReplayer.replayer.peekFront()
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error when checking webhooks: %w", err)
+		}
+
+		if interaction.Type == incomingInteraction {
+			webhookBytes = append(webhookBytes, interaction)
+			_, err = httpReplayer.replayer.popFront()
+			if err != nil {
+				return nil, nil, fmt.Errorf("Unexpectedly reached end of cassette when checking for pending webhooks: %w", err)
+			}
+		} else {
+			break
+		}
+	}
+
+	// --- Deserialize the bytes into HTTP request & response pairs
+	webhookRequests = make([]*httpRequest, 0)
+	webhookResponses = make([]*httpResponse, 0)
+	for _, rawWebhookBytes := range webhookBytes {
+		var reqReader io.Reader = bytes.NewReader(rawWebhookBytes.Request)
+		webhookHTTPRequest, err := httpRequestfromBytes(&reqReader)
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error when deserializing cassette to replay webhooks: %w", err)
+		}
+
+		// TODO: test webhooks
+		webhookRequests = append(webhookRequests, webhookHTTPRequest.(*httpRequest))
+
+		var respReader io.Reader = bytes.NewReader(rawWebhookBytes.Response)
+		webhookHTTPResponse, err := httpResponsefromBytes(&respReader)
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error when deserializing cassette to replay webhooks: %w", err)
+		}
+
+		// Caller of this function is expected to call .Body().Close() on each http.Response
```

Comment:
This comment too.

Reply:
Fixed in 2f60063d2bd1e627c2288aadfe50345bdf191a1e